.github/workflows/auto-release.yml

```diff
@@ -1,60 +1,104 @@
-# Publishes a new release of the package
-# whenever there is a new tagged commit to the master branch.
-#
-# The typical workflow for cutting a new release requires executing the following commands:
-# - git checkout master
-# - bumpversion [major/minor/patch] (pick one)
-# - python setup.py sdist bdist_wheel
-# - twine upload dist/*
-# - git push && git push --tags
-# - rm -rf dist
-# - rm -rf build
-#
-# This action thus eliminates the latter two steps by automating them away,
-# as they are routine and always the same.
-# The `.bumpversion.cfg` file automatically configures git to tag the commit with a release tag
-# and put in a commit message.
-# As such, the new workflow becomes:
-# - git checkout master
-# - bumpversion [major/minor/patch] (pick one)
-# - git push && git push --tags
-# i.e. we save two commands that need to be remembered.
-name: Publish on PyPI
+# Good example resources
+# https://riggaroo.dev/using-github-actions-to-automate-our-release-process/
+# https://blog.eizinger.io/12274/using-github-actions-to-automate-gitflow-style-releases
+
+name: Auto-release
 
 on:
-  push:
-    branches:
-      - master
-  # Executing on PRs just to make sure the first two steps run correctly.
-  pull_request:
-    branches: 
-      - master
+
+  # The below workflow_dispatch section is for a "manual" kick off of the
+  # auto-release script. To cut a new release, navigate to the Actions section
+  # of the repo and select this workflow (Auto-release) on the right hand side.
+  # Then, click "Run workflow" and you will be prompted to input the new
+  # version (which should be major, minor, or patch).
+  workflow_dispatch:
+    inputs:
+      version_name:
+        description: "One of major, minor, or patch"
+        required: true
 
 jobs:
-  publish:
-    name: Publish package
+  release:
+    name: Create a new release
     runs-on: ubuntu-latest
 
     steps:
       - name: Checkout repository
         uses: actions/checkout@v2
 
+      - name: Pull latest commits of dev branch
+        run: |
+          git checkout dev
+          git pull
+
+      # See https://github.com/marketplace/actions/wait-on-check
+      # TODO: Fix this later (have every push to dev trigger a test suite run)
+      # - name: Ensure all tests are passing on dev
+      #   uses: lewagon/wait-on-check-action@v0.2
+      #   with:
+      #     ref: dev  # can be commit SHA or tag
+      #     check-name: pyjanitor tests
```

Comment:
wrong repo 😆

Reply:
Bahhhh, haha, okay, now you know where the thing came from lol.